src/index.tsx

```diff
@@ -207,9 +207,7 @@ export type ServerFormInfo<
       : FormDefInputs[Key] extends { validationAttrs: object }
       ? FormDefInputs[Key]["validationAttrs"] extends { type: "checkbox" }
         ? FormDefInputs[Key]["validationAttrs"] extends { required: true }
-          ? FormDefInputs[Key]["multiple"] extends true
-            ? string[]
-            : string
+          ? string[]
```

Comment:
Checkboxes are weird and currently handled by the lib as sort of an implicit `multiple: true` so updated the types to reflect that (see following commit where I also do it for the non-required case).  Then I also committed the updated `test.tsx` file with the changes to the checkbox inputs.

I may re-think radio/checkboxes in more depth later on, but for now I'm happy with a single-rendered checkbox still reporting an array of checked values.